src/python/test/test_extract_dataset.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 DNAnexus, Inc.
+#
+# This file is part of dx-toolkit (DNAnexus platform client libraries).
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#   use this file except in compliance with the License. You may obtain a copy
+#   of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+from __future__ import print_function, unicode_literals, division, absolute_import
+
+import unittest
+import tempfile
+import shutil
+import os
+import subprocess
+import pandas as pd
+
+class TestDXExtractDataset(unittest.TestCase):
```

Comment:
Great test set! Since there were a couple of exceptions raised in the code (like non-existent output directory) I would suggest to add a (could be single) negative test to verify we indeed fail on that (instead of silently passing).:)

Reply:
Added 1 test case for `file already exists` error. Other exceptions are being tested through QE.